core/src/main/scala/com/metamx/tranquility/druid/DruidBeamMaker.scala

```diff
@@ -82,59 +70,79 @@ class DruidBeamMaker[A: Timestamper](
-        ),
-        new RealtimeIOConfig(
-          new ClippedFirehoseFactory(
-            new TimedShutoffFirehoseFactory(
-              new EventReceiverFirehoseFactory(
-                location.environment.firehoseServicePattern format firehoseId,
-                config.firehoseBufferSize,
-                null
-              ), shutoffTime
-            ), interval
-          ),
-          null
-        ),
-        new RealtimeTuningConfig(
-          druidTuning.maxRowsInMemory,
-          druidTuning.intermediatePersistPeriod,
-          beamTuning.windowPeriod,
-          null,
-          null,
-          if (beamTuning.maxSegmentsPerBeam > 1) {
-            // Experimental setting, can cause tasks to cover many hours. We still want handoff to occur mid-task,
-            // so we need a non-noop rejection policy. Druid won't tell us when it rejects events due to its
-            // rejection policy, so this breaks the contract of Beam.propagate telling the user when events are and
-            // are not dropped. This is bad, so, only use this rejection policy when we absolutely need to.
-            new ServerTimeRejectionPolicyFactory
-          } else {
-            new NoopRejectionPolicyFactory
-          },
-          druidTuning.maxPendingPersists,
-          shardSpec,
-          null,
-          null,
-          null,
-          null,
-          null
-        )
-      )
+    val shardSpecMap = Map(
+      "type" -> "linear",
+      "partitionNum" -> partition
+    ).asJava
+    val queryGranularityMap = druidObjectMapper.convertValue(
+      rollup.indexGranularity,
+      classOf[ju.Map[String, AnyRef]]
     )
+    val dataSchemaMap = Map(
+      "dataSource" -> dataSource,
+      "parser" -> Map(
+        "type" -> "map",
+        "parseSpec" -> Map(
+          "format" -> "json",
+          "timestampSpec" -> timestampSpec,
+          "dimensionsSpec" -> rollup.dimensions.specMap
+        ).asJava
+      ).asJava,
+      "metricsSpec" -> rollup.aggregators.toArray,
+      "granularitySpec" -> Map(
+        "type" -> "uniform",
+        "segmentGranularity" -> beamTuning.segmentGranularity,
+        "queryGranularity" -> queryGranularityMap
+      ).asJava
+    ).asJava
+    val ioConfigMap = Map(
+      "type" -> "realtime",
+      "plumber" -> null,
+      "firehose" -> Map(
+        "type" -> "clipped",
+        "interval" -> interval,
+        "delegate" -> Map(
+          "type" -> "timed",
+          "shutoffTime" -> shutoffTime,
+          "delegate" -> Map(
+            "type" -> "receiver",
+            "serviceName" -> location.environment.firehoseServicePattern.format(firehoseId),
+            "bufferSize" -> config.firehoseBufferSize
+          ).asJava
+        ).asJava
+      ).asJava
+    ).asJava
+    val tuningConfigMap = Map(
+      "type" -> "realtime",
+      "maxRowsInMemory" -> druidTuning.maxRowsInMemory,
```

Comment:
I thought `maxRowsInMemory` was the good one and `rowFlushBoundary` was the deprecated one.